### PR TITLE
added connection with the API

### PR DESCRIPTION
**Pull request by MarcosCGlobant**
In this PR i added the connection to the API, with a preview of the data on the View (showing the name of the first meme of the response on the app).

The result of the implementation is shown with a GIF:

first response

GIF of a smaller device

first response smaller device

### CODE REVIEW

File: app/src/main/java/com/example/memeitupapp/data/service/api/MemeItUpApi.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.example.memeitupapp.data.service.api
+
+import com.example.memeitupapp.data.service.response.MemeBaseResponse
+import com.example.memeitupapp.data.service.response.MemeResponse
+import retrofit2.Call
+import retrofit2.http.GET
+
+interface MemeItUpApi {
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/memeitupapp/data/service/api/MemeItUpApi.kt:8]: Pls Marcos, rename all the classes/interfaces with that name...it is not representative name.

File: app/src/main/java/com/example/memeitupapp/ui/memeitup/MemeuItUpViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class MemeItUpViewModel : ViewModel() {
+
+    private val memesRepository: MemeRepository = MemeRepository()
+    private var mutableMainState: MutableLiveData<Event<Data<List<Meme>>>> = MutableLiveData()
+    val mainState: LiveData<Event<Data<List<Meme>>>>
+        get() {
+            return mutableMainState
```
Comment (benjaglobant) [inline app/src/main/java/com/example/memeitupapp/ui/memeitup/MemeuItUpViewModel.kt:23]: You should use Kotlin style

File: app/src/main/java/com/example/memeitupapp/ui/memeitup/MemeItUpActivity.kt [hunk 3]
```diff
@@ -3,11 +3,33 @@ package com.example.memeitupapp.ui.memeitup
...
 import androidx.appcompat.app.AppCompatActivity
 import com.example.memeitupapp.R
+import com.example.memeitupapp.data.entity.Meme
+import com.example.memeitupapp.util.DEFAULT_INT
+import com.example.memeitupapp.util.Data
+import com.example.memeitupapp.util.Event
+import kotlinx.android.synthetic.main.activity_main.activity_main_call_service_button
+import kotlinx.android.synthetic.main.activity_main.activity_main_first_meme_id
 
 class MemeItUpActivity : AppCompatActivity() {
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/memeitupapp/ui/memeitup/MemeItUpActivity.kt:13]: here the same, in the viewmodel same, rename.

File: app/src/main/java/com/example/memeitupapp/ui/memeitup/MemeuItUpViewModel.kt (hunk 2 above)
Comment (MarcosCGlobant) [reply app/src/main/java/com/example/memeitupapp/ui/memeitup/MemeuItUpViewModel.kt:23]: done, thanks

File: app/src/main/java/com/example/memeitupapp/data/mapper/MemesMapperService.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package com.example.memeitupapp.data.mapper
+
+import com.example.memeitupapp.data.entity.Meme
+import com.example.memeitupapp.data.service.response.MemeResponse
+
+open class MemesMapperService : BaseMapper<MemeResponse, Meme> {
```
[the remark this answers is not in the document]
Comment (MarcosCGlobant) [reply app/src/main/java/com/example/memeitupapp/data/mapper/MemesMapperService.kt:6]: no it doesn't, removed

File: app/src/main/java/com/example/memeitupapp/ui/memeitup/MemeItUpActivity.kt (hunk 3 above)
Comment (MarcosCGlobant) [reply app/src/main/java/com/example/memeitupapp/ui/memeitup/MemeItUpActivity.kt:13]: renamed, thanks

File: app/src/main/java/com/example/memeitupapp/data/service/api/MemeItUpApi.kt (hunk 1 above)
Comment (MarcosCGlobant) [reply app/src/main/java/com/example/memeitupapp/data/service/api/MemeItUpApi.kt:8]: renamed, thanks

File: app/src/main/java/com/example/memeitupapp/ui/gridmemes/GridMemesFragment.kt [hunk 5]
```diff
@@ -6,13 +6,35 @@ import android.view.View
...
 class GridMemesFragment : Fragment() {
 
+    private lateinit var gridMemesViewModel: GridMemesViewModel
+
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? =
         inflater.inflate(R.layout.fragment_grid_memes, container, false)
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        gridMemesViewModel = GridMemesViewModel()
```
Comment (lucianoeli) [inline app/src/main/java/com/example/memeitupapp/ui/gridmemes/GridMemesFragment.kt:25]: this is not the way of intantiate viewmodel. check it please.
i bet your data is lost when configuration changes

Comment (MarcosCGlobant) [reply app/src/main/java/com/example/memeitupapp/ui/gridmemes/GridMemesFragment.kt:25]: fixed, thanks for the help

**Pull request merged**